src/inmanta/env.py

```diff
@@ -676,36 +676,18 @@ def init_env(self) -> None:
         self._parent_python = sys.executable
 
         # check if the virtual env exists
-        if not os.path.exists(self.python_path):
-            # venv requires some care when the .env folder already exists
-            # https://docs.python.org/3/library/venv.html
-            if not os.path.exists(self.env_path):
-                path = self.env_path
-            else:
-                # venv has problems with symlinks
-                path = os.path.realpath(self.env_path)
-
-            # --clear is required in python prior to 3.4 if the folder already exists
-            try:
-                venv.create(path, clear=True, with_pip=False)
-                self._write_pip_binary()
-                self._write_pth_file()
-            except CalledProcessError as e:
-                raise VenvCreationFailedError(msg=f"Unable to create new virtualenv at {self.env_path} ({e.stdout.decode()})")
-            except Exception:
-                raise VenvCreationFailedError(msg=f"Unable to create new virtualenv at {self.env_path}")
-            LOGGER.debug("Created a new virtualenv at %s", self.env_path)
-        elif not os.path.exists(self._path_pth_file):
-            # make sure the venv hosts the same python version as the running process
-            if sys.platform != "win32":
-                # on UNIX based systems, the python version is in the path to the site packages dir:
+        if os.path.isdir(self.env_path) and os.listdir(self.env_path):
+            # Make sure the venv hosts the same python version as the running process
+            if sys.platform.startswith("linux"):
```

Comment:
Could you explain why this needed to be turned around?

Reply:
This was in response to this comment : https://github.com/inmanta/inmanta-core/pull/3973#discussion_r824479544